internal/ingress/controller/template/template.go

```diff
@@ -287,9 +288,26 @@ func buildResolvers(res interface{}, disableIpv6 interface{}) string {
 	return strings.Join(r, " ") + ";"
 }
 
+// ususRewriteAnnotation checks if any location uses the nginx.ingress.kubernetes.io/rewrite-target annotation
+func usesRewriteAnnotation(input interface{}) bool {
```

Comment:
given we do https://github.com/kubernetes/ingress-nginx/pull/3078/files#r217158979, I think renaming this to something like `atLeastOneNeedsRewrite(locations)` would make more sense.